okhttp/src/main/java/okhttp3/internal/connection/Transmitter.kt

```diff
@@ -13,309 +13,300 @@
+    val result = Exchange(this, call, eventListener, exchangeFinder!!, codec)
 
-    synchronized (connectionPool) {
-      this.exchange = result;
-      this.exchangeRequestDone = false;
-      this.exchangeResponseDone = false;
-      return result;
+    synchronized(connectionPool) {
+      this.exchange = result
+      this.exchangeRequestDone = false
+      this.exchangeResponseDone = false
+      return result
     }
   }
 
-  void acquireConnectionNoEvents(RealConnection connection) {
-    assert (Thread.holdsLock(connectionPool));
+  fun acquireConnectionNoEvents(connection: RealConnection) {
+    assert(Thread.holdsLock(connectionPool))
 
-    if (this.connection != null) throw new IllegalStateException();
-    this.connection = connection;
-    connection.transmitters.add(new TransmitterReference(this, callStackTrace));
+    check(this.connection == null)
+    this.connection = connection
+    connection.transmitters.add(TransmitterReference(this, callStackTrace))
   }
 
   /**
    * Remove the transmitter from the connection's list of allocations. Returns a socket that the
    * caller should close.
    */
-  @Nullable Socket releaseConnectionNoEvents() {
-    assert (Thread.holdsLock(connectionPool));
-
-    int index = -1;
-    for (int i = 0, size = this.connection.transmitters.size(); i < size; i++) {
-      Reference<Transmitter> reference = this.connection.transmitters.get(i);
-      if (reference.get() == this) {
-        index = i;
-        break;
-      }
-    }
+  fun releaseConnectionNoEvents(): Socket? {
+    assert(Thread.holdsLock(connectionPool))
 
-    if (index == -1) throw new IllegalStateException();
+    val index = connection!!.transmitters.indexOfFirst { it.get() == this@Transmitter }
+    check(index != -1)
 
-    RealConnection released = this.connection;
-    released.transmitters.remove(index);
-    this.connection = null;
+    val released = this.connection
+    released!!.transmitters.removeAt(index)
+    this.connection = null
 
     if (released.transmitters.isEmpty()) {
-      released.idleAtNanos = System.nanoTime();
+      released.idleAtNanos = System.nanoTime()
       if (connectionPool.connectionBecameIdle(released)) {
-        return released.socket();
+        return released.socket()
       }
     }
 
-    return null;
+    return null
   }
 
-  public void exchangeDoneDueToException() {
-    synchronized (connectionPool) {
-      if (noMoreExchanges) throw new IllegalStateException();
-      exchange = null;
+  fun exchangeDoneDueToException() {
+    synchronized(connectionPool) {
+      check(!noMoreExchanges)
+      exchange = null
     }
   }
 
   /**
-   * Releases resources held with the request or response of {@code exchange}. This should be called
-   * when the request completes normally or when it fails due to an exception, in which case {@code
-   * e} should be non-null.
+   * Releases resources held with the request or response of [exchange]. This should be called when
```

Comment:
you replaced all the `@code`s with `[]`, is it intentional?

Reply:
Yep.